### PR TITLE
Add workflow identifier to all tasks

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Add workflow identifier to all tasks, instead of relying on passing additional namespace ID / workflow ID / run ID
* Modify code accordingly
* Remove deprecated code path

**Why?**
Make logic less coupled

**How did you test it?**
Existing test & CICD

**Potential risks**
N/A

**Is hotfix candidate?**
No

### CODE REVIEW

File: common/persistence/serialization/task_serializer.go [hunk 1]
```diff
@@ -34,28 +34,18 @@ import (
 
 	enumsspb "go.temporal.io/server/api/enums/v1"
 	persistencespb "go.temporal.io/server/api/persistence/v1"
+	"go.temporal.io/server/common/definition"
 	"go.temporal.io/server/common/primitives/timestamp"
 	"go.temporal.io/server/common/tasks"
 )
 
 type (
 	TaskSerializer struct {
```
Comment (yiminc) [inline common/persistence/serialization/task_serializer.go:43]: should this be combined with serialization.Serializer?

**Review by yiminc — APPROVED** [review]
rename suggestion: workflowIdentifier -> WorkflowKey.

**Comment by wxing1292** [comment]
> rename suggestion: workflowIdentifier -> WorkflowKey.

I can certainly do this renaming, but can i do ^ later?

File: common/persistence/serialization/task_serializer.go (hunk 1 above)
Comment (wxing1292) [reply common/persistence/serialization/task_serializer.go:43]: sure, will do in this PR

**Pull request merged**